includes/Core/Modules/Module_With_Blockable_Tags_Trait.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Trait Google\Site_Kit\Core\Modules\Module_With_Blockable_Tags_Trait
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+/**
+ * Trait for a module that has blockable tags.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+trait Module_With_Blockable_Tags_Trait {
+
+	/**
+	 * Checks whether or not the tag should be blocked from rendering.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	protected function is_tag_blocked() {
+		/* @var Module $this The module instance. */
+
+		/**
+		 * Filters whether or not the tag should be blocked from rendering.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param bool $blocked Whether or not the tag output is suppressed. Default: false.
+		 */
+		$blocked = (bool) apply_filters( "googlesitekit_{$this->slug}_tag_blocked", false );
```

Comment:
Right now this filter runs for AMP too, it should only be run for non-AMP. This similar to how we have other filters like `*_init_tag` and `*_init_tag_amp` that run on non-AMP vs AMP respectively.